components/feature/prompts/src/main/java/mozilla/components/feature/prompts/file/FilePicker.kt

```diff
@@ -107,19 +111,24 @@ internal class FilePicker(
      * @param intent The result of the request.
      */
     fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent?) {
-        if (requestCode == FILE_PICKER_ACTIVITY_REQUEST_CODE) {
+        val request = getActivePromptRequest() ?: return
+        if (requestCode == FILE_PICKER_ACTIVITY_REQUEST_CODE && request is File) {
             store.consumePromptFrom(sessionId) {
-                val request = it as File
-
                 if (resultCode == RESULT_OK) {
                     handleFilePickerIntentResult(intent, request)
                 } else {
                     request.onDismiss()
                 }
             }
         }
+        if (request !is File) {
+            logger.debug("Invalid PromptRequest expected File but $request was provided")
```

Comment:
Let's log this as at least `warn` or even `error`. Since we don't know how we end up in this state and really shouldn't...